src/statistics.rs

```diff
@@ -0,0 +1,146 @@
+/// or for a specific stripe.
+#[derive(Debug, Clone)]
+pub struct ColumnStatistics {
+    number_of_values: u64,
+    /// Use aid in 'IS NULL' predicates
+    has_null: bool,
+    type_statistics: Option<TypeStatistics>,
+}
+
+impl ColumnStatistics {
+    pub fn number_of_values(&self) -> u64 {
+        self.number_of_values
+    }
+
+    pub fn has_null(&self) -> bool {
+        self.has_null
+    }
+
+    pub fn type_statistics(&self) -> Option<&TypeStatistics> {
+        self.type_statistics.as_ref()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum TypeStatistics {
+    /// For TinyInt, SmallInt, Int and BigInt
+    Integer {
+        min: i64,
+        max: i64,
+        /// If sum overflows then recorded as None
+        sum: Option<i64>,
+    },
+    /// For Float and Double
+    Double {
+        min: f64,
+        max: f64,
+        /// If sum overflows then recorded as None
+        sum: Option<f64>,
+    },
+    String {
+        min: String,
+        max: String,
+        /// Total length of all strings
+        sum: i64,
+    },
+    /// For Boolean
+    Bucket { true_count: u64 },
+    Decimal {
+        // TODO: use our own decimal type?
+        min: String,
+        max: String,
+        sum: String,
+    },
+    Date {
+        /// Days since epoch
+        min: i32,
+        max: i32,
+    },
+    Binary {
+        // Total number of bytes across all values
+        sum: i64,
+    },
+    Timestamp {
+        /// Milliseconds since epoch
+        /// These were used before ORC-135
+        /// Where local timezone offset was included
+        min: i64,
+        max: i64,
+        /// Milliseconds since UNIX epoch
+        min_utc: i64,
+        max_utc: i64,
+    },
+    Collection {
+        min_children: u64,
+        max_children: u64,
+        total_children: u64,
+    },
+}
+
+impl TryFrom<&proto::ColumnStatistics> for ColumnStatistics {
+    type Error = error::Error;
+
+    fn try_from(value: &proto::ColumnStatistics) -> Result<Self, Self::Error> {
+        let type_statistics = if let Some(stats) = &value.int_statistics {
+            Some(TypeStatistics::Integer {
+                min: stats.minimum(),
+                max: stats.maximum(),
+                sum: stats.sum,
+            })
+        } else if let Some(stats) = &value.double_statistics {
+            Some(TypeStatistics::Double {
+                min: stats.minimum(),
+                max: stats.maximum(),
+                sum: stats.sum,
+            })
+        } else if let Some(stats) = &value.string_statistics {
+            Some(TypeStatistics::String {
+                min: stats.minimum().to_owned(),
+                max: stats.maximum().to_owned(),
+                sum: stats.sum(),
+            })
+        } else if let Some(stats) = &value.bucket_statistics {
+            // TODO: false count?
+            Some(TypeStatistics::Bucket {
+                true_count: stats.count[0], // TODO: safety check this
+            })
```

Comment:
ORC spec is kinda vague on details here, been referring to https://github.com/apache/orc/blob/e1c7e4d9150820fa64b151ad942bffdf81ef34d0/c%2B%2B/src/Statistics.cc#L197

Might need further investigation